### PR TITLE
Builds in C++17 on Ubuntu

### PR DESCRIPTION
**Pull request by dkeeney**
I set the C++ compiler to gcc-8 and set CMake options to not force C++11.
Then cleaned up all of the errors that came up due to compiling under C++17 (without Boost).

There was one error that I could not figure out.  The .py bindings for the function winnerTakesAll_3( ) is broken so I commented it out.  I don't know if this is used by anything.
 I documented it in Issues #177 

So, with the exception of this one .py mapped function, we can now use C++17.
This allows me to now take on Visual Studio 2017.

### CODE REVIEW

**Comment by dkeeney** [comment]
- ok, I fixed the pragma so that the Travis build would work but it is building at C++11 not C++17 because of the age of the compiler.
- CLang version reported by CircleCI is confusing. Apple CLang version 9 is actually CLang 4.  So I need to update my logic to determine if it should support C++17 and filesystem.

**Comment by dkeeney** [comment]
The travis builds but it is building with C++11.  We ask for gcc-8 but GCC 5.4 is actually used.  So something is wrong with the ci script.

**Comment by dkeeney** [comment]
Finally!  C++17 works.
- On travis we have Ubuntu building Release version using C++17.
- On CircleCI we have OSx building a Debug version using C++11.  (The OSx compiler does not support C++17)

Now its time to create a build using Visual Studio 2017 using C++17 (Visual Studio does not support C++11).

**Comment by ctrl-z-9000-times** [comment]
This looks good to me!

**Comment by dkeeney** [comment]
@ctrl-z-9000-times Thanks.
Now on to building on Visual Studio 2017

File: CommonCompilerConfig.cmake [hunk 1]
```diff
@@ -81,17 +81,21 @@ string(TOLOWER ${PLATFORM} PLATFORM)
...
 #    Compiler support for <filesystem> in C++17:
 #	https://en.cppreference.com/w/cpp/compiler_support
+#       https://en.wikipedia.org/wiki/Xcode#Latest_versions
+#
 #	GCC 7.1 has <experimental/filesystem>, link with -libc++experimental or -lstdc++fs
 #	GCC 8 has <filesystem>   link with -lstdc++fs
 #	GCC 9   expected to support <filesystem>
-#	Clang 4 (XCode10) has no support for <filesystem>, partial C++17
+#       AppleClang as of (XCode 10.1) does not support C++17 or filesystem
+#           (although you can get llvm 7 from brew)
```
[the remark this answers is not in the document]
Comment (dkeeney) [reply CommonCompilerConfig.cmake]: > AppleClang as of (XCode 10.1) does not support C++17 or filesystem

So until Apple gets their act together we need to support c++11.

**Pull request merged**